main.js

```diff
@@ -0,0 +1,251 @@
+
+    fetchDailyWeather($currentCity);
+    fetchFiveDayForecast($currentCity);
+  });
+
+  $('.find-me').on('click', function() {
+    $('.processing').css('display', 'block');
+
+    currentCityWeatherData = [];
+    fiveDayCityWeatherData = [];
+    
+    $('.set-default').css('display', 'block');
+
+    if(navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(function(position) {
+          var latitude = position.coords.latitude;
+          var longitude = position.coords.longitude;
+
+          fetchByLocation(latitude, longitude);
+          fetchFiveDayForecastByLocation(latitude, longitude);
+        });
+    } else {
+        alert("Sorry, your location was not found.");
+    }
+  });
+
+  $('#search-query').keypress(function(event) {
+    if (event.keyCode === 13) {
+        $('.search').click();
+    }
+  });
+
+  $('.set-default').on('click', function () {
+      var defaultCity = currentCityWeatherData[0].city;
+      localStorage.setItem('cityName', defaultCity);
+      alert(defaultCity + ' is now your default city.');
+  }); 
+
+  window.onload = function() {
+    var defaultCity = localStorage.getItem('cityName');
+    if (defaultCity !== null) {
+      $('#search-query').val(defaultCity); 
+      $('.search').click();
+      $('.set-default').css('display', 'none');
+    }
+  }
+
+  $(document).ajaxComplete(function () {
+    $('.processing').css('display', 'none');
+  });
+
+  var addCurrentCityWeatherData = function (data) {
+    var currentWeatherImageCode = data.weather[0].icon;
+    var imageSrc = 'http://openweathermap.org/img/wn/' + currentWeatherImageCode +'@2x.png';
+    $('#current-weather-icon').attr('src', imageSrc);
+    var weather_condition = data.weather[0].main;
+
+    var city = data.name;
+    var mapSrc = 'https://www.google.com/maps/embed/v1/place?key=AIzaSyCDyFxxpN5RtNUPm8u2aCKZ8XMqc9mfxdE&q=' + city;
+    $('#map').attr('src', mapSrc);
+    $('.map').css('display', 'block');
+
+    currentCityWeatherData.push({
+      temperature: data.main.temp,
+      city: data.name,
+      weather_conditions: data.weather[0].main,
+    });
+
+    tempToFahrenheit();
+    renderCurrentWeatherCityData();
+    generateBackgroundImage(weather_condition);
+  }
+
+  var addFiveDayCityWeatherData = function (data) {
+    var forecastData = data.list;
+    var forecastDataArray = [];
+    forecastData.map(e => {
+      var temp = e.main.temp_max;
+      var conditions = e.weather[0].main;
+      var weatherIcon = e.weather[0].icon;
+      var date = new Date(e.dt_txt);
+      forecastDataArray.push({ temp: temp, conditions: conditions, weatherIcon: weatherIcon, date: date});
+    });
+
+    convertToDayOfWeek(forecastDataArray);
+    compressWeeklyWeatherData(forecastDataArray);
+  }
+
+  var fetchDailyWeather = function (currentCity) {
+    $.ajax({
+      method: 'GET',
+      url: 'https://api.openweathermap.org/data/2.5/weather?q=' + currentCity + '&appid=345e1c9864ad7ebda8d87ea4d60c53f1',
```

Comment:
I've noticed this `appid=...` is used several times throughout the file. Perhaps it should be stored in a variable so if it were to change you would only have to update it in one place. This is also something that should be set as an environment variable as I mentioned above.